File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    n = len(numbers)
    answer = 0
    
    def dfs(idx,nums):
        if idx == n:
            if nums == target :
                nonlocal answer
                answer += 1
        
        else :
            dfs(idx+1,nums+numbers[idx])
            dfs(idx+1,nums-numbers[idx])
    
    dfs(0,0)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    
    static int answer = 0;
    static int n;
    public int solution(int[] numbers, int target) {
        n = numbers.length;
        
        dfs(0,0,target,numbers);
            
        return answer;
    }
    
    public void dfs(int count , int total, int target,int[] numbers){
        if(count == n){
            if(total == target) answer++;
            return;
        }
        
        dfs(count+1,total+numbers[count],target,numbers);
        dfs(count+1,total-numbers[count],target,numbers);
    }
}